pingcap/tiflow
# checker(dm): port some precheck from lightning

### What problem does this PR solve?


Issue Number: ref https://github.com/pingcap/tiflow/issues/3510

### What is changed and how it works?

port some simple precheck from lightning

### Check List 

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

build a v7.0.0 TiDB

```
command: start-task /tmp/dm_test/lightning_mode/dm-task.yaml --remove-meta "result": true count: 0 != expected: 3
{
    "result": false,
    "msg": "",
    "sources": [
    ],
    "checkResult": "[code=26005:class=dm-master:scope=internal:level=medium], Message: fail to check synchronization configuration with type: check was failed, please see detail
        detail: {
                "results": [
                        {
                                "id": 12,
                                "name": "mysql_version",
                                "desc": "check whether mysql version is satisfied",
                                "state": "warn",
                                "errors": [
                                        {
                                                "severity": "warn",
                                                "short_error": "version suggested less than 8.0.0 but got 8.0.21"
                                        }
                                ],
                                "extra": "address of db instance - 127.0.0.1:3307"
                        },
                        {
                                "id": 24,
                                "name": "lightning_cluster_version",
                                "desc": "check downstream cluster version",
                                "state": "fail",
                                "errors": [
                                        {
                                                "severity": "fail",
                                                "short_error": "Cluster version check failed: [Lightning:Common:ErrVersionMismatch]TiDB version too new, expected to be within [4.0.0, 7.0.0), found '7.0.0'"
                                        }
                                ]
                        }
                ],
                "summary": {
                        "passed": false,
                        "total": 25,
                        "successful": 23,
                        "failed": 1,
                        "warning": 1
                }
        }"
}
```

#### Questions 

##### Will it cause performance regression or break compatibility?

##### Do you need to update user documentation, design documentation or monitoring documentation?

### Release note 

```release-note
`None`.
```
